presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -0,0 +1,253 @@
+        if (domain == null) {
+            // getSummary maybe called before registerQuery
+            return Optional.empty();
+        }
+        else {
+            return domain;
+        }
+    }
+
+    public Supplier<TupleDomain<ColumnHandle>> createDynamicFilterSupplier(QueryId queryId, List<DynamicFilters.Descriptor> dynamicFilters, Map<Symbol, ColumnHandle> columnHandles)
+    {
+        Map<String, ColumnHandle> sourceColumnHandles = extractSourceExpressionSymbols(dynamicFilters, columnHandles);
+
+        return () -> dynamicFilters.stream()
+                .map(filter -> getSummary(queryId, filter.getId())
+                        .map(summary -> translateSummaryToTupleDomain(filter.getId(), summary, sourceColumnHandles)))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .reduce(TupleDomain.all(), TupleDomain::intersect);
+    }
+
+    @Immutable
+    private static class SourceDescriptor
+    {
+        private final QueryId queryId;
+        private final String filterId;
+
+        public static SourceDescriptor of(QueryId queryId, String filterId)
+        {
+            return new SourceDescriptor(queryId, filterId);
+        }
+
+        private SourceDescriptor(QueryId queryId, String filterId)
+        {
+            this.queryId = requireNonNull(queryId, "queryId is null");
+            this.filterId = requireNonNull(filterId, "filterId is null");
+        }
+
+        public QueryId getQueryId()
+        {
+            return queryId;
+        }
+
+        public String getFilterId()
+        {
+            return filterId;
+        }
+
+        @Override
+        public boolean equals(Object other)
+        {
+            if (other == this) {
+                return true;
+            }
+            if (other == null || getClass() != other.getClass()) {
+                return false;
+            }
+
+            SourceDescriptor sourceDescriptor = (SourceDescriptor) other;
+
+            return Objects.equals(queryId, sourceDescriptor.queryId) &&
+                    Objects.equals(filterId, sourceDescriptor.filterId);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(queryId, filterId);
+        }
+
+        @Override
+        public String toString()
+        {
+            return toStringHelper(this)
+                    .addValue(queryId)
+                    .addValue(filterId)
+                    .toString();
+        }
+    }
+
+    private static TupleDomain<ColumnHandle> translateSummaryToTupleDomain(String filterId, Domain summary, Map<String, ColumnHandle> sourceColumnHandles)
+    {
+        if (summary.isNone()) {
+            return TupleDomain.none();
+        }
+        return TupleDomain.withColumnDomains(ImmutableMap.<ColumnHandle, Domain>builder()
+                .put(sourceColumnHandles.get(filterId), summary)
```

Comment:
I think `sourceColumnHandles` can have missing key for `filterId` when dynamic filter is not of `SymbolReference` type. This can happen when you have join key that is composed of two symbols, e.g: `left_x + left_y = right_z`. Then filter will be pushed above table scan that consist of `DF(left_y, left_y, filter_id)`.

Could you add a test case for this?

Reply:
Made changes in `RemoveUnsupportedDynamicFilters` and `DynamicFiltersChecker` to remove dynamic filter when expression is not `SymbolReference` type and verify in plan sanity checker. Added test in `TestDynamicFilter`